src/custom/int.rs

```diff
@@ -250,31 +275,55 @@ mod test {
             })
     }
 
-    #[rstest]
-    fn iadd(mut llvm_ctx: TestContext) {
-        llvm_ctx.add_extensions(add_int_extensions);
-        let hugr = test_binary_int_op("iadd", 3);
-        check_emission!(hugr, llvm_ctx);
+    fn test_unary_int_op(
+        name: impl AsRef<str>,
+        log_width: u8,
+    ) -> Hugr {
+        let ty = &INT_TYPES[log_width as usize];
+        SimpleHugrConfig::new()
+            .with_ins(vec![ty.clone()])
+            .with_outs(vec![ty.clone()])
+            .with_extensions(int_ops::INT_OPS_REGISTRY.clone())
+            .finish(|mut hugr_builder| {
+                let [in1] = hugr_builder.input_wires_arr();
+                let ext_op = int_ops::EXTENSION
+                    .instantiate_extension_op(
+                        name.as_ref(),
+                        [(log_width as u64).into()],
+                        &int_ops::INT_OPS_REGISTRY,
+                    )
+                    .unwrap();
+                let outputs = hugr_builder
+                    .add_dataflow_op(ext_op, [in1])
+                    .unwrap()
+                    .outputs();
+                hugr_builder.finish_with_outputs(outputs).unwrap()
+            })
     }
 
     #[rstest]
-    fn isub(mut llvm_ctx: TestContext) {
+    fn test_neg_emission(mut llvm_ctx: TestContext) {
         llvm_ctx.add_extensions(add_int_extensions);
-        let hugr = test_binary_int_op("isub", 6);
+        let hugr = test_unary_int_op("ineg", 2);
         check_emission!(hugr, llvm_ctx);
     }
 
     #[rstest]
-    fn ieq(mut llvm_ctx: TestContext) {
+    #[case::iadd("iadd", 3)]
+    #[case::isub("isub", 6)]
+    fn test_binop_emission(mut llvm_ctx: TestContext, #[case] op: String, #[case] width: u8) {
         llvm_ctx.add_extensions(add_int_extensions);
-        let hugr = test_binary_icmp_op("ieq", 1);
+        let hugr = test_binary_int_op(op, width);
         check_emission!(hugr, llvm_ctx);
```

Comment:
`check_emission!` uses `insta` to ensure the output doesn't change, but the snapshot names generated by insta don't quite work when using `rstest`'s cases. 

`check_emission!` has a third argument to set a fixed name.
See the float tests:
https://github.com/CQCL/hugr-llvm/blob/ef67c1a7929b1d9d001d56874a64fa86a88792e9/src/custom/float.rs#L258-L263